src/ray/protobuf/gcs_service.proto

```diff
@@ -535,3 +505,34 @@ service PlacementGroupInfoGcsService {
   rpc WaitPlacementGroupUntilReady(WaitPlacementGroupUntilReadyRequest)
       returns (WaitPlacementGroupUntilReadyReply);
 }
+// The following messages/services should never be modified to maintain
+// backward compatibility in autoscaler.
+message GetAllResourceUsageRequest { // Do not change!
+} // Do not change!
+
+message GetAllResourceUsageReply { // Do not change!
+  GcsStatus status = 1; // Do not change!
+  ResourceUsageBatchData resource_usage_data = 2; // Do not change!
+} // Do not change!
+
+// Service for node resource info access.
+service NodeResourceInfoGcsService { // Do not change!
```

Comment:
Hmm I think it is hard to say not to change these services. (it is very much an implementation detail). Can't we just ensure backward compatibility of GetAllResourcesUsage?

Reply:
I am trying to minimize risk. If the name if this service changes, monitor crashes.

Reply:
If someone wants to change it, we can discuss and make sure he doesn’t do something breaking.